### PR TITLE
feat(iOS): Add FileOpenPicker limit for multiple selection

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes https://github.com/unoplatform/kahua-private/issues/145

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs [hunk 1]
```diff
@@ -98,6 +99,37 @@ private void ValidateConfiguration()
...
+	}
+
+	public static class FileOpenPickerExtensions
+	{
+#if __IOS__
+		/// <summary>
+		/// Sets the file limit a user can select when picking multiple files.
+		/// </summary>
+		/// <param name="limit">The maximum number of files that the user can pick.</param>
+		public static void SetLimit(this FileOpenPicker picker, int limit)
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs:117]:
```suggestion
		public static void SetMultipleFilesLimit(this FileOpenPicker picker, int limit)
```

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs [hunk 2]
```diff
@@ -98,6 +99,37 @@ private void ValidateConfiguration()
...
+			picker.SetMultipleFileLimit(limit);
+		}
+
+		/// <summary>
+		/// Shows the file picker so that the user can pick multiple files.
+		/// </summary>
+		/// <param name="limit">The maximum number of files that the user can pick.</param>
+		/// <returns>When the call to this method completes successfully, it returns a <see cref="FilePickerSelectedFilesArray"/> object that contains
+		/// all the files that were picked by the user. Picked files in this array are represented by <see cref="StorageFile"/> objects.</returns>
+		public static IAsyncOperation<IReadOnlyList<StorageFile>> PickMultipleFilesAsync(this FileOpenPicker picker, int limit)
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs:128]: I don't think we need this one, most apps will continue to use the original method, with only a call to `SetMultipleFilesLimit` before that.

I'm still wondering if we should make `SetMultipleFilesLimit` cross targeted already, making it a noop for platforms that don't support it, but with a "notimplemented" attribute on it so developers know it won't work when they use it.

Comment (ajpinedam) [reply src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs]: Removed the method. I will look into making it visible for all platforms but adding the NotImplemented for the rest.

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs [hunk 3]
```diff
@@ -98,6 +99,27 @@ private void ValidateConfiguration()
...
+		/// <summary>
+		/// Sets the file limit a user can select when picking multiple files.
+		/// </summary>
+		/// <param name="limit">The maximum number of files that the user can pick.</param>
+#if !__IOS__
+		[NotImplemented]
 #endif
+		public static void SetMultipleFilesLimit(this FileOpenPicker picker, int limit)
+		{
+			picker.SetMultipleFileLimit(limit);
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs:122]: @ajpinedam This needs to be called only on iOS

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs [hunk 4]
```diff
@@ -98,6 +99,27 @@ private void ValidateConfiguration()
...
+	{
+
+		/// <summary>
+		/// Sets the file limit a user can select when picking multiple files.
+		/// </summary>
+		/// <param name="limit">The maximum number of files that the user can pick.</param>
+#if !__IOS__
+		[NotImplemented]
 #endif
+		public static void SetMultipleFilesLimit(this FileOpenPicker picker, int limit)
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs:120]: Let's also document this in the Windows.Storage docs.

Comment (ajpinedam) [reply src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs:120]: docs were updated

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs (hunk 3 above)
Comment (ajpinedam) [reply src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs:123]: fixed

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.3

**Comment by jeromelaban** [comment]
https://github.com/Mergifyio backport release/stable/5.2

**Pull request merged**